Java/parkservices/src/main/java/com/amazon/randomcutforest/parkservices/PredictorCorrector.java

```diff
@@ -705,10 +709,12 @@ protected <P extends AnomalyDescriptor> P detect(P result, RCFComputeDescriptor
         boolean candidate = false;
 
         if (workingGrade > 0 && lastDescriptor != null) {
-            if (score > lastDescriptor.getRCFScore()
-                    || lastDescriptor.getRCFScore() - lastDescriptor.getThreshold() > score
-                            - max(workingThreshold, lastDescriptor.getThreshold())
-                                    * (1 + max(0.2, runLength / (2.0 * max(10, shingleSize))))) {
+            if (score > lastDescriptor.getRCFScore()) {
+                candidate = true;
+            }
+            double runDiscount = max(workingThreshold, lastDescriptor.getThreshold())
```

Comment:
Do you want to add an else as you only want to do the following when score <= lastDescriptor.getRCFScore() ?

Reply:
Actually no -- and that relates to the comment below (fair point about opacity). Even for a single anomaly what often happens is that the score is high as soon as we see the value -- then score continues to increase till about middle of the shingle (all these are very unlikely patterns) and then the score falls off. So the score actually can keep rising even though no fresh anomaly has been seen. However what happens is that law of marginal increase takes over -- the jump in score when we first see the anomaly is typically higher, and unless there are subsequent anomalies then the marginal increase in score is less as the anomaly moves inward. The threshold is however a lagging computation and peaks much later. So this seemed to eliminate the effect of raised scores due to single anomalies -- but there can be other ways.

Reply:
Fair point. A single execution path may also (i) help debug as well as (ii) decide if the second part is truly important.
Changed.